jump/src/installer.rs

```diff
@@ -231,22 +232,33 @@ pub(crate) fn install(files: &[FileEntry], payload: &[u8]) -> Result<(), String>
                 0
             }
             FileEntry::ScieTote((tote_file, entries)) => {
-                let scie_tote_tmpdir = tempfile::TempDir::new().map_err(|e| {
-                    format!(
-                        "Failed to create a temporary directory to extract the scie-tote to: {e}"
-                    )
-                })?;
-                let scie_tote_path = scie_tote_tmpdir.path().join(&tote_file.name);
-                let bytes = &payload[location..(location + tote_file.size)];
-                unpack(
-                    tote_file.file_type,
-                    tote_file.executable.unwrap_or(false),
-                    || Ok((Cursor::new(bytes), ())),
-                    tote_file.hash.as_str(),
-                    &scie_tote_path,
-                )?;
+                let mut scie_tote: Option<TempDir> = None;
+                let mut scie_tote_src = || {
+                    if let Some(tempdir) = scie_tote.as_ref() {
+                        return Ok::<_, String>(tempdir.path().join(&tote_file.name));
+                    }
+                    let scie_tote_tmpdir = TempDir::new().map_err(|e| {
```

Comment:
Here's the scie tote extraction. To a temp dir, just like before, but this time lazily in a closure that uses an Option to hold a single extraction.

Reply:
Yep, okay, that's what I thought - just wanted to confirm my understanding about the eager extraction bug.